MockServer/src/main/java/com/github/dedis/student20_pop/mockserver/WebSocketServer.java

```diff
@@ -55,13 +55,12 @@ public void onOpen(Session session) {
         public void onMessage(String message, Session session) {
             Pattern pattern = Pattern.compile(ID_REGEX);
             Matcher matcher = pattern.matcher(message);
-            String id = matcher.group(0);
-            if(id != null) {
+            if(matcher.find()) {
                 try {
                     session.getBasicRemote().sendText(
                             String.format(Locale.US,
                                     JSONRPC_FORMAT,
-                                    Integer.parseInt(id)));
+                                    Integer.parseInt(matcher.group(1))));
```

Comment:
What does the 1 represents ? Would it be clearer if it was in a variable ?

Reply:
It's the first match. It's how to use the matcher, like a 0 in an array to get the first element.